client/utils/deep-compare.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Given two objects, deep compares them using JSON.stringify
+ *
+ * @param  {Object}  obj1 Object to compare
+ * @param  {Object}  obj2 Other object to compare
+ * @return {Boolean}      Are both objects completely equal?
+ */
+export default function deepCompare(obj1, obj2) {
```

Comment:
Would [`_.deepEqual`](http://underscorejs.org/#isEqual) work well for this? I expect it would be more efficient, and we already have underscore as a dependency anyways.